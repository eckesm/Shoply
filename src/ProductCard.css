.ProductCard {
	border: 1px solid black;
	width: 300px;
	margin: 10px;
}

.ProductCard-details {
	text-align: left;
	margin: 25px;
}

.ProductCard-detail {
	margin: 0px;
}

.ProductCard-image_url {
	max-height: 250px;
	width: auto;
	max-width: 250px;
}

.ProductCard-buttons {
	margin: 5px;
}
