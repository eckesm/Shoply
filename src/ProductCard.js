import React from 'react';
import { useDispatch } from 'react-redux';
import { addProduct, removeProduct } from './actions';
import './ProductCard.css';

const ProductCard = ({ id, name, price, description, image_url }) => {
	const dispatch = useDispatch();
	const addToCart = () => dispatch(addProduct(id, name));
	const removeFromCart = () => dispatch(removeProduct(id));

	return (
		<div className="ProductCard">
			<h1>{name}</h1>
			<div className="ProductCard-details">
				<p className="ProductCard-detail">
					<b>Price:</b> {price}
				</p>
				<p className="ProductCard-detail">
					<b>Description:</b> {description}
				</p>
			</div>
			<img className="ProductCard-image_url" src={image_url} alt={name} />
			<div className='ProductCard-buttons'>
				<button onClick={addToCart}>Add to Cart</button>
				<button onClick={removeFromCart}>Remove from Cart</button>
			</div>
		</div>
	);
};

export default ProductCard;
