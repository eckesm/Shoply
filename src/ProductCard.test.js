import React from 'react';
import ProductCard from './ProductCard';
import { render, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer from './rootReducer';

const store = createStore(reducer);

it('renders without crashing', () => {
	render(
		<Provider store={store}>
			<ProductCard />
		</Provider>
	);
});

it('passes snapshot test', () => {
	expect(
		render(
			<Provider store={store}>
				<ProductCard
					id="test-id"
					name="test-name"
					price="1.99"
					description="test-description"
					image_url="https://images.samsung.com/is/image/samsung/latin_en-hd-j4300ah-un32j4300ahxpa-001-front-indigo-blue"
				/>
			</Provider>
		)
	).toMatchSnapshot();
});