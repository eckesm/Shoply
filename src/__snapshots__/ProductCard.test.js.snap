// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passes snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ProductCard"
      >
        <h1>
          test-name
        </h1>
        <div
          class="ProductCard-details"
        >
          <p
            class="ProductCard-detail"
          >
            <b>
              Price:
            </b>
             
            1.99
          </p>
          <p
            class="ProductCard-detail"
          >
            <b>
              Description:
            </b>
             
            test-description
          </p>
        </div>
        <img
          alt="test-name"
          class="ProductCard-image_url"
          src="https://images.samsung.com/is/image/samsung/latin_en-hd-j4300ah-un32j4300ahxpa-001-front-indigo-blue"
        />
        <div
          class="ProductCard-buttons"
        >
          <button>
            Add to Cart
          </button>
          <button>
            Remove from Cart
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ProductCard"
    >
      <h1>
        test-name
      </h1>
      <div
        class="ProductCard-details"
      >
        <p
          class="ProductCard-detail"
        >
          <b>
            Price:
          </b>
           
          1.99
        </p>
        <p
          class="ProductCard-detail"
        >
          <b>
            Description:
          </b>
           
          test-description
        </p>
      </div>
      <img
        alt="test-name"
        class="ProductCard-image_url"
        src="https://images.samsung.com/is/image/samsung/latin_en-hd-j4300ah-un32j4300ahxpa-001-front-indigo-blue"
      />
      <div
        class="ProductCard-buttons"
      >
        <button>
          Add to Cart
        </button>
        <button>
          Remove from Cart
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
